### DISCUSSION TITLE
Help for build the frimware for Pinecil im use a WSL Ubuntu

### DISCUSSION THREAD

**Discussion by dedsecproyt** [Ralim/IronOS#1152] [Q&A]
root@DEDSEC-PC:~/IronOS/source# make -j8 model=Pinecil firmware-UK
Building for Pine64

Building file: ./Core/BSP/Pine64/Vendor/SoC/gd32vf103/Common/Source/GCC/startup_gd32vf103.S
Building file: ./Core/BSP/Pine64/Vendor/OS/FreeRTOS/Source/portable/GCC/portasm.S
Compiling ./Core/Src/freertos.c
Compiling ./Core/Src/syscalls.c
Building file: ./Core/BSP/Pine64/Vendor/SoC/gd32vf103/Common/Source/GCC/intexc_gd32vf103.S
Compiling ./Core/Src/FreeRTOSHooks.c
cc1: error: ./Core/Drivers/usb-pd/include: No such file or directory [-Werror=missing-include-dirs]
cc1: error: ./Core/Drivers/usb-pd/include: No such file or directory [-Werror=missing-include-dirs]
Compiling ./Core/BSP/Pine64/Vendor/SoC/gd32vf103/Common/Source/Stubs/gettimeofday.c
Compiling ./Core/BSP/Pine64/Vendor/SoC/gd32vf103/Common/Source/Stubs/isatty.c
cc1: error: ./Core/Drivers/usb-pd/include: No such file or directory [-Werror=missing-include-dirs]
cc1: error: ./Core/Drivers/usb-pd/include: No such file or directory [-Werror=missing-include-dirs]
cc1: all warnings being treated as errors
cc1: error: ./Core/Drivers/usb-pd/include: No such file or directory [-Werror=missing-include-dirs]
make: *** [Makefile:409: Objects/Pinecil/./Core/Src/syscalls.o] Error 1
make: *** Waiting for unfinished jobs....
Compiling ./Core/BSP/Pine64/Vendor/SoC/gd32vf103/Common/Source/Stubs/read.c
cc1: all warnings being treated as errors
cc1: all warnings being treated as errors
cc1: error: ./Core/Drivers/usb-pd/include: No such file or directory [-Werror=missing-include-dirs]
make: *** [Makefile:411: Objects/Pinecil/./Core/BSP/Pine64/Vendor/SoC/gd32vf103/Common/Source/Stubs/isatty.o] Error 1
make: *** [Makefile:411: Objects/Pinecil/./Core/Src/freertos.o] Error 1
cc1: all warnings being treated as errors
make: *** [Makefile:411: Objects/Pinecil/./Core/Src/FreeRTOSHooks.o] Error 1
cc1: all warnings being treated as errors
make: *** [Makefile:411: Objects/Pinecil/./Core/BSP/Pine64/Vendor/SoC/gd32vf103/Common/Source/Stubs/gettimeofday.o] Error 1
cc1: all warnings being treated as errors
make: *** [Makefile:411: Objects/Pinecil/./Core/BSP/Pine64/Vendor/SoC/gd32vf103/Common/Source/Stubs/read.o] Error 1

**Comment by Ralim** [comment]
You did not use a recursive clone when pulling the source code.

You can run `git submodule update` in the folder and it should grab the submodule.

**Reply by dedsecproyt** [reply]
(ironos-venv) dedsec-pro@DEDSEC-PC:~/IronOS/source$ git submodule update
(ironos-venv) dedsec-pro@DEDSEC-PC:~/IronOS/source$ sudo make -j8 model=Pinecil firmware-UK
Building for Pine64

Compiling ./Core/Src/syscalls.c
Compiling ./Core/Src/FreeRTOSHooks.c
Compiling ./Core/BSP/Pine64/Vendor/SoC/gd32vf103/Common/Source/Stubs/gettimeofday.c
Compiling ./Core/Src/freertos.c
Compiling ./Core/BSP/Pine64/Vendor/SoC/gd32vf103/Common/Source/Stubs/isatty.c
cc1: error: ./Core/Drivers/usb-pd/include: No such file or directory [-Werror=missing-include-dirs]
cc1: error: ./Core/Drivers/usb-pd/include: No such file or directory [-Werror=missing-include-dirs]
cc1: error: ./Core/Drivers/usb-pd/include: No such file or directory [-Werror=missing-include-dirs]
cc1: error: ./Core/Drivers/usb-pd/include: No such file or directory [-Werror=missing-include-dirs]
cc1: all warnings being treated as errors
cc1: all warnings being treated as errors
Compiling ./Core/BSP/Pine64/Vendor/SoC/gd32vf103/Common/Source/Stubs/read.c
Compiling ./Core/BSP/Pine64/Vendor/SoC/gd32vf103/Common/Source/Stubs/sbrk.c
Compiling ./Core/BSP/Pine64/Vendor/SoC/gd32vf103/Common/Source/Stubs/lseek.c
make: *** [Makefile:411: Objects/Pinecil/./Core/Src/syscalls.o] Error 1
make: *** Waiting for unfinished jobs....
make: *** [Makefile:411: Objects/Pinecil/./Core/BSP/Pine64/Vendor/SoC/gd32vf103/Common/Source/Stubs/isatty.o] Error 1
cc1: error: ./Core/Drivers/usb-pd/include: No such file or directory [-Werror=missing-include-dirs]
cc1: error: ./Core/Drivers/usb-pd/include: No such file or directory [-Werror=missing-include-dirs]
cc1: error: ./Core/Drivers/usb-pd/include: No such file or directory [-Werror=missing-include-dirs]
cc1: all warnings being treated as errors
make: *** [Makefile:411: Objects/Pinecil/./Core/BSP/Pine64/Vendor/SoC/gd32vf103/Common/Source/Stubs/sbrk.o] Error 1
cc1: all warnings being treated as errors
cc1: all warnings being treated as errors
make: *** [Makefile:411: Objects/Pinecil/./Core/BSP/Pine64/Vendor/SoC/gd32vf103/Common/Source/Stubs/gettimeofday.o] Error 1
make: *** [Makefile:411: Objects/Pinecil/./Core/Src/FreeRTOSHooks.o] Error 1
cc1: error: ./Core/Drivers/usb-pd/include: No such file or directory [-Werror=missing-include-dirs]
cc1: all warnings being treated as errors
make: *** [Makefile:411: Objects/Pinecil/./Core/BSP/Pine64/Vendor/SoC/gd32vf103/Common/Source/Stubs/read.o] Error 1
cc1: all warnings being treated as errors
make: *** [Makefile:411: Objects/Pinecil/./Core/BSP/Pine64/Vendor/SoC/gd32vf103/Common/Source/Stubs/lseek.o] Error 1
cc1: all warnings being treated as errors
make: *** [Makefile:411: Objects/Pinecil/./Core/Src/freertos.o] Error 1

**Reply by Ralim** [reply]
Try `git submodule init && git submodule update` if you didn't have them initialised.
Can you also make sure your checking out the code to a path in WSL that is not the windows remap (just in case)

**Reply by dedsecproyt** [reply]
All good. And can I create a settings backup after flash frimware??

**Reply by Ralim** [reply]
I'm not sure what you want to backup settings for, there isnt many.
You can read the entire flash out to take a snapshot of the firmware + settings.
Otherwise settings are roughly linked to the firmware version so its best to let the firmware decide if it needs to reset them or not
